force-app/main/default/classes/NextStepFuture.cls

```diff
@@ -58,7 +75,9 @@ public with sharing class NextStepFuture {
 
 
     // Helper method to update role name
-    private static void updateUserRoleName(UserRole[] roles){
+    @future
+    private static void updateUserRoleName(){
+        UserRole[] roles = [SELECT Id, Name FROM UserRole WHERE Id = :UserInfo.getUserRoleId() OR Name = 'CRO' LIMIT 1];
```

Comment:
- Doesn't affect the code or test but I'm curious why you moved the UserRole query from outside the method to inside the scope. I assume you had a good reason, just a curiosity that stood out